catalog/ui/search-ui/standard/src/main/webapp/js/widgets/cesium.mapclustering.js

```diff
@@ -0,0 +1,178 @@
+                var polygons = view.polygons;
+                $.each(polygons, function(index, value) {
+                    value.fill.show = true;
+                    value.outline.show = true;
+                });
+            }
+        },
+        hideEntity : function(billboard, view) {
+            billboard.show = false;
+            if(typeof view.polygons !== "undefined" ) {
+                var polygons = view.polygons;
+                $.each(polygons, function(index, value) {
+                    value.fill.show = false;
+                    value.outline.show = false;
+                });
+            }
+        },
+        fitsInExistingCluster: function(position) {
+            var that = this;
+            var cluster = false;
+            $.each(clusters, function(index, value) {
+                if(that.isInRadius(that.getRadius(value), position, value)) {
+                    cluster = value;
+                    return false;
+                }
+            });
+            return cluster;
+        },
+        createNewCluster : function (billboard, view) {
+            var points = [];
+            var views = [];
+            points.push(billboard.position);
+            views.push(view);
+            clusters.push({ref : billboard, views : views, position : billboard.position, points: points, radius : DEFAULT_PIXEL_DISTANCE});
+        },
+        addPointToCluster: function(cluster, billboard, view) {
+            var points = cluster.points;
+            var views = cluster.views;
+            points.push(billboard.position);
+            views.push(view);
+            var boundingSphere = Cesium.BoundingSphere.fromPoints(points);
+            var center = boundingSphere.center;
+            cluster.position = center;
+        },
+        removeClusters : function () {
+            $.each(clusters, function(index, cluster) {
+                viewer.entities.remove(cluster.entity);
+            });
+            clusters = [];
+        },
+        resetBillboard: function()  {
+            var that = this;
+            $.each(clusters, function(index, cluster) {
+                viewer.entities.remove(cluster.entity);
+                $.each(cluster.views, function(index, view) {
+                    that.showEntity(view.billboard, view);
+                });
+            });
+            clusters = [];
+        },
+        isInRadius: function isInRadius(radius, entity, centerEntity) {
+            if(typeof entity.position === "undefined" || typeof centerEntity === "undefined" ) {
+                return false;
+            }
+            var distance = Cesium.Cartesian3.distance(entity.position, centerEntity.position);
+            return (distance < radius);
+        },
+        getRadius: function getRadius(entity) {
+            var width = viewer.canvas.width;
+            var height = viewer.canvas.height;
+            var point1 = viewer.camera.pickEllipsoid(new Cesium.Cartesian2(width * 40 / 100, height / 2));
```

Comment:
could you change these magic numbers to constants, or move the math to well-named method?